sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -410,6 +413,14 @@ object PushProjectionThroughUnion extends Rule[LogicalPlan] with PredicateHelper
       } else {
         p
       }
+
+    case p @ Project(projectList, watermark: EventTimeWatermark) =>
+      // Push as long as the project doesn't eliminates the attribute.
```

Comment:
Do we still need `Project` when `projectList` are identical to the output of `EventTimeWatermark`, except the metadata?